### PR TITLE
Beta: report usage of raw_request

### PR DESCRIPTION
**Pull request by richardm-stripe**
Analogous to https://github.com/stripe/stripe-python/pull/1191.

### CODE REVIEW

File: src/main/java/com/stripe/net/LiveStripeResponseGetter.java [hunk 1]
```diff
@@ -185,7 +185,8 @@ public StripeResponse rawRequest(RawApiRequest apiRequest) throws StripeExceptio
       }
     }
 
-    StripeResponse response = httpClient.requestWithRetries(request);
```
Comment (richardm-stripe) [inline src/main/java/com/stripe/net/LiveStripeResponseGetter.java:188]: This was from a bad merge

File: src/main/java/com/stripe/net/RawApiRequest.java [hunk 2]
```diff
@@ -1,23 +1,51 @@
...
-    this.rawContent = rawContent;
-    this.options = options;
+    this(baseAddress, method, path, options, apiMode, null, rawContent);
+  }
+
+  public RawApiRequest addUsage(String usage) {
+    List<String> newUsage = new ArrayList<>();
+    if (this.getUsage() != null) {
+      newUsage.addAll(this.getUsage());
+    }
```
Comment (anniel-stripe) [inline src/main/java/com/stripe/net/RawApiRequest.java:40]: nit: Maybe add a test to check that existing usage is retained?

**Review by anniel-stripe — APPROVED** [review]

File: src/main/java/com/stripe/net/RawApiRequest.java (hunk 2 above)
Comment (richardm-stripe) [reply src/main/java/com/stripe/net/RawApiRequest.java:40]: Done in [Add test for .addUsage](https://github.com/stripe/stripe-java/pull/1731/commits/ccdb9f10a2c69dc7a00898968f1a12672aba6f44)

File: src/test/java/com/stripe/functional/StripeClientTest.java [hunk 3]
```diff
@@ -45,4 +46,29 @@ public void testReportsUsageTelemetry() throws StripeException {
...
+          JsonArray usage =
+              new Gson()
+                  .fromJson(
+                      stripeRequest.headers().firstValue("X-Stripe-Client-Telemetry").get(),
+                      JsonObject.class)
+                  .get("last_request_metrics")
+                  .getAsJsonObject()
+                  .get("usage")
+                  .getAsJsonArray();
+          System.out.println("usage: " + usage);
```
Comment (pakrym-stripe) [inline src/test/java/com/stripe/functional/StripeClientTest.java]: rem

Comment (richardm-stripe) [inline src/test/java/com/stripe/functional/StripeClientTest.java:68]:
```suggestion
```

**Pull request merged**